client/app/config/services.yml

```diff
@@ -103,3 +103,11 @@ services:
             $storage: '@AppBundle\Service\File\Storage\S3Storage'
             $siriusApiGatewayClient: '@AppBundle\Service\Client\Sirius\SiriusApiGatewayClient'
             $restClient: '@AppBundle\Service\Client\RestClient'
+
+    DateTime:
+        class: 'DateTime'
```

Comment:
This was really weird...I couldn't get Symfony to start without defining DateTime as a service to inject into the fake clock class. Does anyone know if there's any way round this?

Reply:
You're on the right lines with the date provider so you can test it, but you shouldn't need the `FakeClock`. I've used a similar service in api called `DateTimeProvider` which is a service that wraps `new \DateTime`. I inject that into the whatever needs to generate a date, and to test it I just mock that service and tell it to return a hard coded date in the tests.

Reply:
That's much simpler - I've dropped all the clock classes and switched to using the DateTimeProvider instead 👍